feat: adds idea board app with some challanges (#2)
contentcal-org/ccal-tech-test: front-end/README.md

1. Run `npm install`
2. Run `npm start`
3. Enter `http://localhost:4200`

### Running unit tests

- Run `npm run affected:test` to execute the unit tests via Jest on affected files only.
- Run `npm run affected:test --all` to execute the unit tests via Jest on entire project.

### Running linters

- Run `npm run affected:lint` to execute linter on affected files only.
- Run `npm run affected:lint --all` to execute linter on entire project.

## Challange

The Front-end challenge is to improve an idea board where you can create, view, update and delete ideas.
There is a starter app built using NX or you can choose you use your own boilerplate.

![idea-board](https://user-images.githubusercontent.com/8656036/112143530-47478b80-8bd8-11eb-9747-1ce0814fc942.png)

### Challange 1:

Oops, seems like you can access our idea list even if you are not logged in just by entering `http://localhost:4200/ideas`.
This module should be accessed only by logged-in users. Could you please protect it somehow?

_Tip_: State will hold information about logged-in user. You can find `isAuthenticated` selector in `libs/core/src/lib/_states/auth.state.ts` that you can use.

### Challange 2:

> One of our clients have hundreds or thousends of great ideas. Printing so long list of idea have very bad inpact on Angular performance.

Reviewer: ```suggestion
One of our clients has hundreds or thousands of great ideas. Printing such a long list of ideas has a very bad impact on Angular performance.
```

Could you please take a look on `IdeasListingComponent` (`apps/idea-board/src/app/ideas/ideas-list/_components/ideas-listing`) and check

> if we can easly optimalise this component performance?

Reviewer: ```suggestion
if we can easily optimise this component performance?
```
